guide/index.adoc

```diff
@@ -0,0 +1,929 @@
+This makes it possible to define your job as a configurable Docker image and avoid requiring the creation of new Docker images for common changes.
+
+==== Privilege Minimization
+Docker images often are set to use the `root` user by default. 
+Avoid this unless absolutely necessary. 
+Instead of using the `root` user, add a non-privileged user to the Docker image and set the {docker-run-user-url}[USER] for the image to this non-privileged user.
+
+==== Dockerfiles
+Docker provides guidance on some additional {docker-best-practices-url}[best practices] for writing Dockerfiles.
+
+
+=== Frequently Asked Questions (FAQs)
+
+==== What is this Seed image?
+{seed-url}[Seed] is a general {seed-url}/seed.html[standard] that was developed to aid in the discovery and running of self-contained algorithms. 
+It is a JSON-defined manifest that is embedded into the Docker image as a label. 
+It defines:
+
+	* Algorithm - name, version, description
+	* Developer - name, company, contact info
+	* Inputs/Outputs - how many, what type
+	* Hardware Requirements - CPUs, memory, disk
+	* Error handling - data vs. algorithm errors
+	* Environment variables
+	* Mounts
+
+==== How do I use Docker?
+Many resources for learning about Docker can be found on the https://www.docker.com[Docker website]. 
+The primary goal of this guide is to explain how to integrate an algorithm with Seed and not focus on how to build a Docker container.
+
+==== Why not just add all the files in my directory to the Docker image?
+The image should be as small as necessary. It is pulled across the network each time it runs on a node.  
+Smaller images result in less network traffic.
+
+==== Why not just increase the hardware resources as much as possible?  I want it to run fast.
+It is critical to strike a balance between the minimum resources required to run the algorithm and acceptable runtimes. 
+Scale depends on accurate requirements in order to efficiently schedule jobs on the cluster. 
+If a job requests more resources than it uses, these resources are wasted as they cannot not be allocated to other jobs. 
+In a cloud environment, wasted resources equate to wasted money. 
+When job requirements depend on the input, the Seed manifest supports specifying a multiplicative value for the resource requests via the `inputMultiplier` property.
+
+==== My algorithm requires some reference data, should I just embed it into the Docker image?
+It depends. If the total size of the file(s) is sufficiently small; e.g. less than 100 MB, then add it into the Docker image. 
+If it has larger requirements, such as a lookup directory, please work with the Scale team to create a mountable directory.
+
+==== How to secure sensitive information like passwords inside my Docker image?
+The Seed manifest allows the developer to define "Settings", which will be defined as environment variables at runtime. 
+These "Settings" can be marked as "secret" and will be stored in {vault-url}[HashiCorp Vault]. 
```

Comment:
Should probably mention this is Scale specific. Seed doesn't require any vault setup.

Reply:
That makes sense, updated in [df17722](https://github.com/ngageoint/seed-cli/pull/233/commits/df1772214e5dfcaf9c9bf7d76daa0a62fda86b0c)